src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1109,29 +1059,33 @@ void Ekf2::run()
 					}
 				}
 
-				// Check and save the last valid calibration when we are disarmed
-				if ((vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_STANDBY)
-				    && (status.filter_fault_flags == 0)
+				// Check and update the last valid magnetometer calibration
```

Comment:
@priseborough Can you please quickly sanity-check the logic here? What this now does is that it gets the _updated_ mag bias from the EKF at every step (this is for correcting the raw mag readings in `sensor_corrected`). The actual parameter _save_ is still only done when disarmed.